tests/xbot/common/controls/sensors/XboxControllerTest.java

```diff
@@ -0,0 +1,27 @@
+package xbot.common.controls.sensors;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import xbot.common.controls.sensors.XXboxController.XboxButton;
+import xbot.common.injection.BaseWPITest;
+import xbot.common.injection.wpi_factories.WPIFactory;
+import xbot.common.logging.RobotAssertionException;
+
+public class XboxControllerTest extends BaseWPITest {
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+    
+    @Test
+    public void doubleAllocateButton() {
+        WPIFactory factory = injector.getInstance(WPIFactory.class);
+        XXboxController controller = factory.getXboxController(0);
+        
+        // We expect the robot to get mad if we try to get the same button twice
+        thrown.expect(RobotAssertionException.class);
```

Comment:
Ah, nevermind, I see how it works now. Yep, that's cleaner.

I think this rule approach is meant when you want to enable/disable checking during the test itself.